coco_architecture.md

```diff
@@ -0,0 +1,147 @@
+
+![CC_TEE_container](https://github.com/confidential-containers/documentation/blob/main/images/CC_TEE_container.jpg)
+
+Following is the workflow when deploying a confidential container (Kubernetes POD):
+
+- CC workload preparation
+  - User builds the container image(s)  (podman build -f …)
+  - User signs/encrypts the container image(s)
+  - User pushes the container image(s) to the image registry (podman push …)
+- Deploying the CC workload in k8s
+  - User deploys the workload (kubectl apply -f <...>)
+  - Kubernetes schedules the workload to target host having the required capability to run confidential containers
+- CC workload execution flow (**red connector** in the diagram)
+  - Confidential containers runtime on the host starts the enclave (VM TEE)
+  - Enclave (agent) performs remote attestation: **steps 1-2 from pic**
+  - Enclave (agent) gets the keys required to verify/decrypt the containers image(s):  **steps 3-4 from the pic**
+  - Enclave (image management) downloads the container image(s) : **step 5 from the pic**
+  - Enclave verifies/decrypts the container image(s) : **step 6 from the pic**
+  - Enclave starts the container workload
+
+## Typical Confidential Container Stack using process based TEE (SGX)
+
+The following diagram shows how a CC solution is used to provide a confidential process to the end user: 
+
+![CC_SGX_container](https://github.com/confidential-containers/documentation/blob/main/images/CC_SGX_container.jpg)
+
+Following is the workflow when deploying a confidential container (Kubernetes POD). 
+We have marked in blue the changes from the confidential container using VM based TEE scenario: 
+
+- CC workload preparation
+  - User builds the container image(s)  (podman build -f …)
+  - User signs/encrypts the container image(s)
+  - User pushes the container image(s) to the image registry (podman push …)
+- Deploying the CC workload in k8s
+  - User deploys the workload (kubectl apply -f <...>)
+  - Kubernetes schedules the workload to target host having the required capability to run confidential containers
+- CC workload execution flow (**red connector** in the diagram)
+  - Confidential containers runtime on the host starts the enclave (SGX) 
+  - Enclave (agent) performs remote attestation: **steps 1-2 from pic**
+  - Enclave (agent) gets the keys required to verify/decrypt the containers image(s):  **steps 3-4 from the pic**
+  - Enclave (image management) downloads the container image(s) : **step 5 from the pic**
+  - Enclave verifies/decrypts the container image(s) : **step 6 from the pic**
+  - Enclave starts the container workload
+
+As can be seen, the flows for process and VM based TEEs are almost identical. It should be noted that process based TEE 
+requires a few additional software components like libOS without which the application requires re-architecting. 
+This is not the case for VM based TEEs.
+
+
+## Typical Confidential VM Stack
```

Comment:
Do we want to have this section at all? A "typical" full blown confidential computing VM stack can be quite complex (OVMF, kernel, initrd, disk image, etc) and also not as well/statically defined as the confidential containers one. So instead of providing a super generic diagram, I'd suggest to just remove that section.